scylla/tests/integration/query_result.rs

```diff
@@ -295,3 +307,131 @@ async fn execute_iter_serialization_error() {
+            );
+            assert_eq!(structured_history.requests.len(), 1);
+            running_proxy
+        },
+    )
+    .await;
+
+    match res {
+        Ok(()) => (),
+        Err(err) => panic!("{}", err),
+    }
+}
+
+#[tokio::test]
+#[ntest::timeout(60000)]
+#[cfg(not(scylla_cloud_tests))]
+async fn test_session_should_have_metadata() {
+    setup_tracing();
+    let session = create_new_session_builder().build().await.unwrap();
+    let state = session.get_cluster_state();
+    let keys = ["SCYLLA_URI", "SCYLLA_URI2", "SCYLLA_URI3"];
+    let expected_addresses: HashSet<String> = keys
+        .iter()
+        .map(|key| env::var(key).unwrap_or_else(|_| panic!("{} not set", key)))
+        .collect();
+
+    let got_addresses: HashSet<String> = state
+        .get_nodes_info()
+        .iter()
+        .map(|node| node.address.to_string())
+        .collect();
+
+    assert_eq!(
+        got_addresses, expected_addresses,
+        "Cluster node addresses do not match environment variables"
+    );
+}
+
+#[tokio::test]
+async fn test_should_not_have_tracing_id_when_tracing_disabled() {
+    setup_tracing();
+    let session = create_new_session_builder().build().await.unwrap();
+    let query: Query = Query::new("SELECT release_version FROM system.local");
+
+    let result = session.query_unpaged(query, &[]).await.unwrap();
+    let tracing_id: Option<Uuid> = result.tracing_id();
+    assert!(tracing_id.is_none());
+}
+
+#[tokio::test]
+async fn test_should_fetch_trace_when_tracing_enabled() {
+    let session = create_new_session_builder().build().await.unwrap();
+
+    let mut query = Query::from("SELECT release_version FROM system.local");
+    query.set_tracing(true);
+
+    let result = session.query_unpaged(query, &[]).await.unwrap();
+    let tracing_id: Option<Uuid> = result.tracing_id();
+    assert!(tracing_id.is_some());
+    let tracing_info: TracingInfo = session
+        .get_tracing_info(&tracing_id.unwrap())
+        .await
+        .unwrap();
+
+    // Verify trace information is present and has expected format
+    assert!(!tracing_info.events.is_empty());
+
+    // Check if the request type matches
+    assert_eq!(tracing_info.request.unwrap(), "Execute CQL3 query");
+
+    // Verify duration is positive
+    assert!(tracing_info.duration.unwrap() > 0);
+
+    // Verify started_at timestamp is present
+    assert!(tracing_info.started_at.unwrap().0 > 0);
+
+    // Check parameters
+    assert!(tracing_info
+        .parameters
+        .clone()
+        .unwrap()
+        .contains_key("consistency_level"));
+    assert!(tracing_info.parameters.unwrap().contains_key("query"));
+
+    // Check events
+    for event in tracing_info.events {
+        assert!(!event.activity.clone().unwrap().is_empty());
+        assert!(event.source.is_some());
+        assert!(event.source_elapsed.unwrap() >= 0);
+        assert!(!event.activity.unwrap().is_empty());
+    }
+}
```

Comment:
Could you look into `test_tracing_` tests in `integration/session.rs`? It looks to me like those tests mostly duplicate the ones already existing - maybe instead of adding new ones you could improve existing ones with the checks that you wrote?